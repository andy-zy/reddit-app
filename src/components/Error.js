// @flow
import React, { Component } from 'react'
import {
  Text,
  View,
} from 'react-native'

import type { PropsBaseT } from '../domain/types'

import { layout, colors } from '../styles'

export default class Error extends Component<PropsBaseT> {

  static navigationOptions = {
    title: 'Error',
  }

  render() {
    const { navigation } = this.props
    const error: string = navigation.state.params.error || 'Something went wrong!'

    return (
      <View style={layout.vContainer}>
        <Text style={colors.error}>{error}</Text>
      </View>
    )
  }
}