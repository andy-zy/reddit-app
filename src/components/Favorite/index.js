import Favorite from './Favorite.container'

export default Favorite