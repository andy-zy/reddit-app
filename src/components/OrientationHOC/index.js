import OrientationHOC from './OrientationHOC'

export default OrientationHOC