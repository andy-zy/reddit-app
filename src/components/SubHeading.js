// @flow
import React from 'react'
import { Text } from 'react-native'

import { layout } from '../styles'

export default (props: Object) => <Text {...props} style={[layout.subHeading, props.style]}>{props.children}</Text>