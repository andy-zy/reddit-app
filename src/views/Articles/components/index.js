import Tabs from './Tabs'
import Search from './Search'
import ArticleList from './ArticleList'
import ListItem from './ListItem'

export {
  Tabs,
  Search,
  ArticleList,
  ListItem,
}