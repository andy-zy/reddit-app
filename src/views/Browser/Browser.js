// @flow
import React from 'react'
import {
  WebView,
} from 'react-native'

import type { RenderT } from './types'

const Browser = ({ activeArticle }: RenderT) => <WebView source={{ uri: activeArticle.url }} />

export default Browser