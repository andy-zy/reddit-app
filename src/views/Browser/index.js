import Browser from './Browser.container'

export default Browser