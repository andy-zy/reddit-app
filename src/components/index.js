import { default as Favorite } from './Favorite'
import { default as OrientationHOC } from './OrientationHOC'

export {
  Favorite,
  OrientationHOC,
}