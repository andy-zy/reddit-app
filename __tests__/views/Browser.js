import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import Browser from '../../src/views/Browser/Browser'
import { Favorite } from '../../src/components'

describe('Error', () => {
  let props

  beforeEach(() => {
    props = {
      activeArticle: { url: 'https://test.com/' },
    }
  })

  it('renders correctly', () => {
    const tree = renderer.create(<Browser {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('navigationOptions headerRight is <Favorite />', () => {
    expect(Browser.navigationOptions.headerRight).toMatchObject(<Favorite />)
  })
})
