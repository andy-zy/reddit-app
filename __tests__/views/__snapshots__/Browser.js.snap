// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTWebView
    allowsInlineMediaPlayback={undefined}
    automaticallyAdjustContentInsets={undefined}
    bounces={undefined}
    contentInset={undefined}
    dataDetectorTypes={undefined}
    decelerationRate={undefined}
    injectedJavaScript={undefined}
    mediaPlaybackRequiresUserAction={undefined}
    messagingEnabled={false}
    onLoadingError={[Function]}
    onLoadingFinish={[Function]}
    onLoadingStart={[Function]}
    onMessage={[Function]}
    onShouldStartLoadWithRequest={undefined}
    scalesPageToFit={true}
    scrollEnabled={undefined}
    source={
      Object {
        "uri": "https://test.com/",
      }
    }
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "backgroundColor": "#ffffff",
        },
        undefined,
      ]
    }
  />
</View>
`;
